### PR TITLE
create delete route,destroy functions in UserController and UserServi…

### PR DESCRIPTION
**Pull request by AKV85**
1) prepared endpoint DELETE '/users/{id}'
2) only the owner  be able to delete the account
3) added the 'status' int field to the 'users' table (created new migration for that)

by default, this value  be equal to 1 (active)
4) added constants to the User class:
const ACTIVE = 1;
const INACTIVE = 2;
5) when the user will send a delete request,
      - his status in the 'users' table  changed to 2 (inactive) use constants.
      -  generated pdf file with any content 
      -  pdf file sent to his email

### CODE REVIEW

**Comment by AKV85** [comment]
Screenshot 2023-06-09 131212
Screenshot 2023-06-09 131146

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -43,4 +46,13 @@ public function update(UserUpdateRequest $request, User $user): JsonResponse
 
         return response()->json($updatedUser);
     }
+
+    public function destroy(DeleteUserRequest $request, User $user)
+    {
+        $request->authorize();
```
Comment (Pavel-Tsymbal) [inline app/Http/Controllers/UserController.php]: don't call it, it runs automatically

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -2,24 +2,27 @@
...
 use App\Http\Resources\UserResource;
 use App\Models\User;
 use App\Services\UserService;
+use Barryvdh\DomPDF\Facade\Pdf;
 use Symfony\Component\HttpFoundation\JsonResponse;
 
 class UserController extends Controller
 {
     protected UserService $userService;
+    protected $pdf;
```
Comment (Pavel-Tsymbal) [inline app/Http/Controllers/UserController.php]: remove it

File: app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -2,24 +2,27 @@
...
 
-    public function __construct(UserService $userService)
+    public function __construct(UserService $userService, PDF $pdf)
```
Comment (Pavel-Tsymbal) [inline app/Http/Controllers/UserController.php]: remove $pdf

File: app/Http/Controllers/UserController.php [hunk 4]
```diff
@@ -2,24 +2,27 @@
...
     {
         $this->userService = $userService;
+        $this->pdf = $pdf;
```
Comment (Pavel-Tsymbal) [inline app/Http/Controllers/UserController.php]: move pdf generating to the service method

**Pull request merged**